pkg/controller/nodes/executor.go

```diff
@@ -406,8 +406,9 @@ func (c *nodeExecutor) handleNode(ctx context.Context, w v1alpha1.ExecutableWork
 		logger.Errorf(ctx, "failed Execute for node. Error: %s", err.Error())
 		return executors.NodeStatusUndefined, err
 	}
+
 	execErr := p.GetErr()
-	if execErr != nil && p.GetPhase() == handler.EPhaseRetryableFailure && nodeStatus.GetLastAttemptStartedAt() != nil {
+	if execErr != nil && nodeStatus.GetLastAttemptStartedAt() != nil {
```

Comment:
Add a comment here:

// Err object should not be populated for other phase, and we only reach here for v1alpha1.NodePhaseQueued, v1alpha1.NodePhaseRunning

Also comment on lint 401 is incorrect. Remove v1alpha1.NodePhaseRetryableFailure in line 401